src/main/java/org/dimdev/dimdoors/mixin/PlayerEntityMixin.java

```diff
@@ -18,18 +21,73 @@
 import net.minecraft.util.Identifier;
 import net.minecraft.world.World;
 
+import java.util.Random;
+
 @Mixin(value = PlayerEntity.class, priority = 900)
 public abstract class PlayerEntityMixin extends LivingEntity {
+	private static final int RANDOM_ACTION_BOUND = 400;
+	private static final int CHANCE_TO_DECREASE_ARMOR_DURABILITY = 20;
+	private static final int CHANCE_TO_REPLACE_ITEMSLOT_WITH_UNRAVLED_FABRIC = 10;
+	Random random = new Random();
+
 	@Shadow
 	public abstract void incrementStat(Identifier stat);
 
 	public PlayerEntityMixin(EntityType<? extends LivingEntity> entityType, World world) {
 		super(entityType, world);
 	}
 
+	@Inject(method = "tick", at = @At("HEAD"), cancellable = true)
```

Comment:
Move all of your new code into ServerPlayerEntityMixin instead and sync item/ slot changes via Packets. Anything that is random should happen exclusively server-side and be synced to the client.

Reply:
i uh, i changed over the stuff, but I didn't have to use packets?? idunno if I did it correctly or not, I don't know how packets or server side stuff works, but I just called the functions in the serverplayerentity mixin and everything still works?